src/gt4py/next/common.py

```diff
@@ -141,6 +141,13 @@ def __and__(self, other: Set[Any]) -> UnitRange:
         else:
             raise NotImplementedError("Can only find the intersection between UnitRange instances.")
 
+    def __le__(self, other: Set[Any]):
+        if isinstance(other, UnitRange):
+            # required for infinity comparison
```

Comment:
as discussed: the default implementation will loop over the pseudo-infinite range

Reply:
Well it would also be a performance nightmare. I don't think any operation should actually use the abc implementation, but that can be addressed in the fixup PR.

Reply:
yes exactly